nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java

```diff
@@ -5018,9 +5018,11 @@ public void verifyCanUpdate(final VersionedFlowSnapshot updatedFlow, final boole
             final Map<String, VersionedConnection> proposedConnections = new HashMap<>();
             findAllConnections(updatedFlow.getFlowContents(), proposedConnections);
 
-            findAllConnections().stream()
-                    .filter(conn -> conn.getVersionedComponentId().isPresent())
-                    .forEach(conn -> proposedConnections.remove(conn.getVersionedComponentId().get()));
+            for (Connection conn : flowManager.findAllConnections()) {
```

Comment:
Same comment as above. Here, I think we need to use the existing `findAllConnections()` method. We only want to change to `flowManager.findAllConnections()` when calling it on the root process group.